test/format-impl-test.cc

```diff
@@ -353,6 +353,19 @@ TEST(format_impl_test, count_digits) {
   test_count_digits<uint64_t>();
 }
 
+// Tests fmt::detail::countl_zero for unsigned type UInt.
+template <typename UInt> void test_countl_zero() {
+  constexpr auto num_bits = fmt::detail::num_bits<UInt>();
+  UInt n = 1u;
+  for (int i = 1; i < num_bits - 1; i++) {
+    n <<= 1;
+    EXPECT_EQ(fmt::detail::countl_zero(n - 1), num_bits - i);
+    EXPECT_EQ(fmt::detail::countl_zero(n), num_bits - i - 1);
+  }
+}
```

Comment:
Do we need this template? Why not merge the logic into the test and get rid of it?

Reply:
Merged.